libraries/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -791,74 +942,28 @@ void _IotMqtt_ProcessSend( IotTaskPool_t pTaskPool,
         {
             pOperation->u.operation.status = IOT_MQTT_NETWORK_ERROR;
         }
-        else
+        /* DISCONNECT operations are considered successful upon successful transmission. */
+        else if( pOperation->u.operation.type == IOT_MQTT_DISCONNECT )
         {
-            /* DISCONNECT operations are considered successful upon successful
-             * transmission. In addition, non-waitable operations with no callback
-             * may also be considered successful. */
-            if( pOperation->u.operation.type == IOT_MQTT_DISCONNECT )
-            {
-                /* DISCONNECT operations are always waitable. */
-                IotMqtt_Assert( waitable == true );
+            /* DISCONNECT operations are always waitable. */
+            IotMqtt_Assert( waitable == true );
 
-                pOperation->u.operation.status = IOT_MQTT_SUCCESS;
-            }
-            else if( waitable == false )
+            pOperation->u.operation.status = IOT_MQTT_SUCCESS;
+        }
+        /* Non-waitable operations with no callback are also considered successful. */
+        else if( waitable == false )
+        {
+            if( pOperation->u.operation.notify.callback.function == NULL )
             {
-                if( pOperation->u.operation.notify.callback.function == NULL )
-                {
-                    pOperation->u.operation.status = IOT_MQTT_SUCCESS;
-                }
```

Comment:
This if block had a nesting level of 5, so the outermost `else` was refactored into `else if`s to reduce the nesting depth.